src/mito2/src/schedule/remote_job_scheduler.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use common_telemetry::error;
+use store_api::storage::RegionId;
+use tokio::sync::mpsc::Sender;
+
+use crate::compaction::compactor::CompactionRegion;
+use crate::compaction::picker::PickerOutput;
+use crate::error::{Error, Result};
+use crate::manifest::action::RegionEdit;
+use crate::request::{BackgroundNotify, CompactionFailed, CompactionFinished, WorkerRequest};
+
+pub type RemoteJobSchedulerRef = Arc<dyn RemoteJobScheduler>;
+
+/// RemoteJobScheduler is a trait that defines the API to schedule remote jobs.
+#[async_trait::async_trait]
+pub trait RemoteJobScheduler: Send + Sync + 'static {
+    /// Sends a job to the scheduler and returns a unique identifier for the job.
+    async fn schedule(&self, job: RemoteJob, notifier: Arc<dyn Notifier>) -> Result<JobId>;
+}
+
+/// Notifier is used to notify the mito engine when a remote job is completed.
+#[async_trait::async_trait]
+pub trait Notifier: Send + Sync + 'static {
+    /// Notify the mito engine that a remote job is completed.
+    async fn notify(&self, result: RemoteJobResult);
+}
+
+/// JobId is a unique identifier for a remote job and allocated by the scheduler.
+#[derive(Default, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct JobId(u64);
```

Comment:
Maybe we can use uuid.

Reply:
Could we use the `Uuid` struct directly? `String` isn't copyable but the `Uuid` struct is.

Reply:
The `ProcedureId` is an example
https://github.com/GreptimeTeam/greptimedb/blob/8702066967bd3a68896b55cdee6ccde3eeca8812/src/common/procedure/src/procedure.rs#L264-L266